modules/solid_mechanics/src/materials/IsotropicPowerLawHardening.C

```diff
@@ -0,0 +1,74 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "IsotropicPowerLawHardening.h"
+
+#include "SymmIsotropicElasticityTensor.h"
+
+#include "PiecewiseLinear.h"
+
+/**
+* Isotropic power law hardening material model. Before yield, the stress is youngs modulus * strain.
+* After yielding, the stress is K* pow(strain, n) where K is the strength coefficient,  n is the strain
+* hardening exponent and strain is the total strain.
+* Yield stress is the point of intersection of these two curves.
+**/
+
+template<>
+InputParameters validParams<IsotropicPowerLawHardening>()
+{
+  InputParameters params = validParams<IsotropicPlasticity>();
+
+  // Linear strain hardening parameters
+  params.set<Real>("yield_stress") = 1.0;
```

Comment:
Oh yeah, that makes sense.  The yield stress and hardening constant constant would show up as parameters, so a user could set them and erroneously think they were used. Maybe this shouldn't really be derived from IsotropicPlasticity, or maybe you shouldn't pull in its parameters.